internal/cli/docker_prune.go

```diff
@@ -0,0 +1,101 @@
+package cli
+
+import (
+	"context"
+	"time"
+
+	"github.com/opentracing/opentracing-go"
+	"github.com/spf13/cobra"
+
+	"github.com/windmilleng/tilt/internal/docker"
+	"github.com/windmilleng/tilt/internal/minikube"
+
+	"github.com/windmilleng/tilt/internal/k8s"
+	"github.com/windmilleng/tilt/pkg/logger"
+)
+
+type dockerPruneCmd struct {
+	untilStr string
+}
+
+func (c *dockerPruneCmd) register() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "docker-prune",
+		Short: "run docker prune as tilt does",
+	}
+
+	cmd.Flags().StringVar(&c.untilStr, "until", "12h", "max age of image to keep (as go duration string, e.g. 1h30m, 12h")
+
+	return cmd
+}
+
+func (c *dockerPruneCmd) run(ctx context.Context, args []string) error {
+	// a := analytics.Get(ctx)
+	// a.Incr("cmd.dockerPrune", map[string]string{
+	// 	"watch": fmt.Sprintf("%v", c.watch),
+	// 	"mode":  string(updateModeFlag),
+	// })
+	// a.IncrIfUnopted("analytics.up.optdefault")
+	// defer a.Flush(time.Second)
+
+	span, ctx := opentracing.StartSpanFromContext(ctx, "Up")
+	defer span.Finish()
+
+	// tags := tracer.TagStrToMap(c.traceTags)
+
+	// for k, v := range tags {
+	// 	span.SetTag(k, v)
+	// }
+
+	deferred := logger.NewDeferredLogger(ctx)
+	ctx = redirectLogs(ctx, deferred)
+
+	dCli, err := provideDockerClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	until, err := time.ParseDuration(c.untilStr)
+	if err != nil {
+		return err
+	}
+
+	err = dCli.Prune(ctx, until)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func provideDockerClient(ctx context.Context) (docker.Client, error) {
```

Comment:
hmmm...did you write this by hand? can you use the wireDockerClusterClient that `tilt doctor` uses?

Reply:
oh yea, I just hacked this file together to test with. will clean this up!